ops/testing.py

```diff
@@ -14,8 +14,11 @@
 
 """Infrastructure to build unit tests for charms using the ops library."""
 
+import typing as _typing
+from importlib.metadata import PackageNotFoundError as _PackageNotFoundError
+from importlib.metadata import version as _get_package_version
```

Comment:
It was because I couldn't get Sphinx to properly work with `__all__` when using `automodule`. However, I had to give up using `automodule` anyway, because I couldn't find a clean way to get it to use `ops.testing` rather than `ops._private.harness` and `scenario` for references.

So that means that `__all__` is actually viable - I'll go back to that. I'll also rely on `__all__` defining the API and not bother importing `typing as _typing` and so on.